zerver/lib/typed_endpoint_validators.py

```diff
@@ -21,12 +22,16 @@ def check_string_fixed_length(string: str, length: int) -> Optional[str]:
     return string
 
 
-def check_int_in(val: int, possible_values: List[int]) -> int:
+def check_int_in(val: int, possible_values: Iterable[int]) -> int:
     if val not in possible_values:
         raise ValueError(_("Not in the list of possible values"))
     return val
 
 
+def check_int_in_after_validator(possible_values: Iterable[int]) -> AfterValidator:
+    return AfterValidator(lambda val: check_int_in(val, possible_values))
```

Comment:
I'm not sure the naming is ideal for these validators, but the coding pattern seems nicer than what we were doing before! 

@PIG208 FYI in case you have ideas for an even better approach.

Reply:
I'd prefer shorter names like `check_int_in_validator` instead of `check_int_in_after_validator`.

This validation approach should be fine for this PR.

For checks of whether an integer is within a list of constants, having `IntEnum` types will make the parameter type annotations shorter and eliminate the need for validators. We have done something similar before -- see https://github.com/zulip/zulip/pull/24618/commits/b13e319c081cae1b076060a7af7aae8744359569. The downside is that it will likely require a lot more work.

Reply:
> For checks of whether an integer is within a list of constants, having `IntEnum` types will make the parameter type annotations shorter and eliminate the need for validators.

The problem with `IntEnum` is that we cannot pass an array to it, we would have to specify all the values. This wouldn't work for arrays that are used in multiple places. Unless we change all those instances to use the `IntEnum`.

Reply:
I have changed the name to `check_int_in_validator`

Reply:
What do you mean? The point of the `IntEnum` suggestion is that you don’t need to pass an array, or use a custom validator at all; Pydantic already knows what the possible values are.

Reply:
What i mean is when defining the `IntEnum` we cannot pass an existing list. 
For example, `UserProfile.COLOR_SCHEME_CHOICES` is a `List`, if we were to define an `IntEnum` we would have to specify all the choices manually.
```
class ColorSchemeEnum:
    color1 = 1
    color2 = 2
    ....
```
This would mean that if in the future we decide to modify the `UserProfile.COLOR_SCHEME_CHOICES` list we would have to manually modify the Enum as well, unless we change `UserProfile.COLOR_SCHEME_CHOICES` and all its uses to an `Enum`(which would require a lot of changes as said by @PIG208 ).

Which approach would you suggest?

Reply:
> This would mean that if in the future we decide to modify the UserProfile.COLOR_SCHEME_CHOICES list we would have to manually modify the Enum as well, unless we change UserProfile.COLOR_SCHEME_CHOICES and all its uses to an Enum

Yeah. Because full `IntEnum` proposal requires a lot of work, it is best done in the future separately.

For this PR we can stick with the validator approach.

Reply:
Yeah let's plan to do `IntEnum` later, once we've gotten the `typed_endpoint` sweep finished. One thing to note is that most of the settings currently using an enum-type integer field are going to be migrated to group-setting value permissions as part of #19525, so we may benefit from being lazy about doing refactoring of those to `IntEnum` fields.

Reply:
I guess we could prototype changing the `corporate/` views to use `IntEnum in parallel with migrating additional files; @kennethnrk are you up for attempting that?